x-pack/plugin/async-search/src/test/java/org/elasticsearch/xpack/search/RestSubmitAsyncSearchActionTests.java

```diff
@@ -88,7 +89,7 @@ public void testParameters() throws Exception {
             assertThat(request, instanceOf(SubmitAsyncSearchRequest.class));
             assertThat(valueAccessor.apply((SubmitAsyncSearchRequest) request), equalTo(expectedValue));
             executeCalled.set(true);
-            return null;
+            return new AsyncSearchResponse("", randomBoolean(), randomBoolean(), 0L, 0L);
```

Comment:
This was pretty dirty to begin with. Without this change and the change in the test verifying client we would always throw an NPE in the background (form trying to serialize a `null` response)  but that would never fail any tests.
With the new assertions that wasn't possible anymore so I had to fix this up here.